app/helpers/ethnic_background_helper.rb

```diff
@@ -48,4 +48,16 @@ def ethnic_background_title(group)
   def equality_and_diversity_caption
     '<span class="govuk-caption-xl">Equality and diversity</span>'.html_safe
   end
+
+  # return every possible combination of group + background
+  # used in the factory for generating sensible combinations in test data
+  def all_combinations
```

Comment:
`ETHNIC_GROUPS.product(ETHNIC_BACKGROUNDS)`? https://ruby-doc.org/core-2.6.6/Array.html#method-i-product

Reply:
I did initially try that, but couldn't get it to produce the output I needed (an array-of-arrays but no further nesting) without the code ending up at least as complex as doing it explicitly (`ETHNIC_BACKGROUNDS` is a Hash, so you have to dereference it to the particular group, and you can't do it as a straightforward map or you end up with 3-level-deep nesting....

That's not to say it's not possible, of course, just that I didn't want to waste too much time on it

Reply:
ah, I didn't realise it was a hash! trying to be clever, please ignore me

Reply:
I got nerd sniped by this:

```ruby
ETHNIC_BACKGROUNDS
  .to_a
  .map {|(group, bgs)| ([group] * bgs.size).zip(bgs) }
  .flatten(1)
``` 

Works the same, but is vastly more arcane than the suggested code... yeah, stick to the nested loop!